### PR TITLE
Add meaningful error message before crashing when bundle identifier cant be found

### LINKED ISSUE #46: An invalid identifier produces a bogus error
This is the response:

```
$ bundle exec crep crashes --identifier='hello.world' --version='7.12.0' --build='10038' --only-unresolved
/crep/lib/crep/model/crash_sources/hockeyapp_crash_source.rb:17:in `configure': undefined method `title' for nil:NilClass (NoMethodError)
	from /crep/crash_controller.rb:9:in `initialize'
	from /crep/lib/crep/command/crashes.rb:42:in `new'
	from /crep/lib/crep/command/crashes.rb:42:in `run'
	from /.rvm/gems/ruby-2.3.1/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
	from /crep/lib/crep.rb:13:in `run'
	from /crep/bin/crep:7:in `<top (required)>'
	from /.rvm/gems/ruby-2.3.1/bin/crep:23:in `load'
	from /.rvm/gems/ruby-2.3.1/bin/crep:23:in `<main>'
	from /.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
	from /.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
```

I expect that [this](https://github.com/xing/crep/blob/master/lib/crep/model/crash_sources/hockeyapp_crash_source.rb#L73) is `nil`, as the array is empty. We should throw an error if that is the case.

### PR DESCRIPTION
**Pull request by dhartwich1991**
Adds a log before exiting from the program so you know what went wrong. See the screenshot below.
Fixes #46

terminal

### CODE REVIEW

File: lib/crep/model/crash_sources/hockeyapp_crash_source.rb [hunk 1]
```diff
@@ -52,7 +52,10 @@ def url(app_id:, version_id:, reason_id:)
     def version(version:, build:)
       filtered_versions = filtered_versions_by_version_and_build(@hockeyapp_app.versions, version, build)
 
-      raise "No version was found for #{version})#{build})".red unless filtered_versions.count > 0
+      if filtered_versions.count <= 0
+        CrepLogger.error("No version was found for #{version})#{build})")
+        raise
```
Comment (BasThomas) [inline lib/crep/model/crash_sources/hockeyapp_crash_source.rb]: We decided (for now) to raise after logging the error as it wouldn't make sense to continue. It does feel like a two-step "fix"/process which can be easily forgotten.

The problem with moving `raise` to the Logger is that the logger will never return.

What do you think @q231950? Is this a worthy "workaround" for now?

Comment (q231950) [reply lib/crep/model/crash_sources/hockeyapp_crash_source.rb]: I think _"it wouldn't make sense to continue"_ is a good reason to make a last meaningful log and then fail the execution 👍

File: lib/crep/model/crash_sources/hockeyapp_crash_source.rb [hunk 2]
```diff
@@ -70,6 +73,10 @@ def hockeyapp(bundle_identifier, client)
       apps = all_apps.select do |app|
         app.bundle_identifier == bundle_identifier
       end
+      if !apps.first
+        CrepLogger.error("No app with the give bundle identifier (#{bundle_identifier}) was found.") unless apps.first
```
Comment (q231950) [inline lib/crep/model/crash_sources/hockeyapp_crash_source.rb]: I think `unless apps.first` is useless because you check for `!apps.first` just before 😉

Comment (BasThomas) [reply lib/crep/model/crash_sources/hockeyapp_crash_source.rb]: Nice catch!

**Comment by q231950** [comment]
@dhartwich1991 this looks like it can be merged 😉

**Pull request merged**